ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/util/CommandParser.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.tools.migrations.util;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.confluent.ksql.execution.expression.tree.BooleanLiteral;
+import io.confluent.ksql.execution.expression.tree.CreateArrayExpression;
+import io.confluent.ksql.execution.expression.tree.CreateMapExpression;
+import io.confluent.ksql.execution.expression.tree.CreateStructExpression;
+import io.confluent.ksql.execution.expression.tree.DecimalLiteral;
+import io.confluent.ksql.execution.expression.tree.DoubleLiteral;
+import io.confluent.ksql.execution.expression.tree.Expression;
+import io.confluent.ksql.execution.expression.tree.IntegerLiteral;
+import io.confluent.ksql.execution.expression.tree.LongLiteral;
+import io.confluent.ksql.execution.expression.tree.StringLiteral;
+import io.confluent.ksql.metastore.TypeRegistry;
+import io.confluent.ksql.parser.AstBuilder;
+import io.confluent.ksql.parser.DefaultKsqlParser;
+import io.confluent.ksql.parser.KsqlParser;
+import io.confluent.ksql.parser.tree.InsertValues;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public final class CommandParser {
+  private static final String QUOTED_STRING_OR_WHITESPACE = "('([^']*|(''))*')|\\s+";
+  private static final KsqlParser KSQL_PARSER = new DefaultKsqlParser();
+  private static final String INSERT = "INSERT";
+  private static final String INTO = "INTO";
+  private static final String VALUES = "VALUES";
+  private static final String SELECT = "SELECT";
+  private static final String CREATE = "CREATE";
+  private static final String SINK = "SINK";
+  private static final String SOURCE = "SOURCE";
+  private static final String CONNECTOR = "CONNECTOR";
+  private static final String SHORT_COMMENT_OPENER = "--";
+  private static final String SHORT_COMMENT_CLOSER = "\n";
```

Comment:
Do we have integration tests / have we run manual tests with files with newlines? It'd be good to sanity-check that carriage returns don't pose a problem.

Reply:
Will add this to follow-up PR.